src/redux/books/books.js

```diff
@@ -1,50 +1,75 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
 const ADD_BOOK = 'bookstoreapp/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstoreapp/books/REMOVE_BOOK';
+const GET_BOOK = 'bookstoreapp/books/GET_BOOK';
 
-const initialState = [
-  {
-    title: 'JavaScript ES6',
-    author: 'Ankole',
-    id: 1,
+const apiKey = 'kqkewfRoZMkq2lXZaEMG';
+const baseUrl = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${apiKey}/books`;
+export const addBook = createAsyncThunk(
+  (ADD_BOOK),
+  async (payload) => {
+    const postData = await axios.post(
+      baseUrl, payload,
+    );
+    return postData;
   },
-  {
-    title: 'React-Redux ',
-    author: 'Anko',
-    id: 2,
+);
+
+export const deleteBooks = createAsyncThunk(
+  (REMOVE_BOOK),
+  async (id) => {
+    const response = await axios.delete(`${baseUrl}/${id}`);
+    return response;
   },
-  {
-    title: 'TDD in React',
-    author: 'Ankoe',
-    id: 3,
+);
+
+export const getBooks = createAsyncThunk(
+  (GET_BOOK),
+  async () => {
+    const response = await axios.get(baseUrl);
+    const data = await response.data;
+    return data;
   },
-];
+);
 
-export const addBook = (id, title, author) => ({
-  type: ADD_BOOK,
-  id,
-  title,
-  author,
-});
+export const bookSlice = createSlice({
+  name: 'bookData',
+  initialState: {
+    books: [],
+    status: null,
+  },
+  reducers: {
+    removeBook(state, action) {
+      state.books = state.books.filter(
+        (book) => book.item_id !== action.payload.item_id,
+      );
+    },
 
```

Comment:
While you did good implementing this function whenever I add a book its working fine but when i click the remove button it does not seem to work it would be nice if you corrected this issue

Reply:
I also noticed in you categories section when i click the check status button its supposed to say under construction but your check status button isnt functional please kindly fix this issue.